internal/oc/extract.go

```diff
@@ -0,0 +1,50 @@
+package oc
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/openshift/assisted-service/pkg/executer"
+	"github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen -source=extract.go -package=oc -destination=mock_extract.go
+type Extracter interface {
+	Extract(log logrus.FieldLogger, imageIndexPath string, openshiftVersion string, filePath string, pullSecret string) (string, error)
+	ExtractDatabaseIndex(log logrus.FieldLogger, openshiftVersion string, pullSecret string) (string, error)
+}
+
+type extract struct {
+	executer executer.Executer
+	config   Config
+}
+
+func NewExtracter(executer executer.Executer, config Config) Extracter {
+	return &extract{executer, config}
+}
+
+const (
+	imageIndex           = "registry.redhat.io/redhat/redhat-operator-index"
+	dbFile               = "/database/index.db"
+	templateImageExtract = "oc image extract %s:v%s --path=%s:%s"
```

Comment:
Should support `--insecure=` flag like on `release.go` for mirroring
https://github.com/openshift/assisted-service/blob/687ed5f0379139a448075b17ae2c2714515a10c3/internal/oc/release.go#L50-L51